smart-contracts/advanced/deterministic-deployment.md

```diff
@@ -0,0 +1,141 @@
+
+3. **Broadcast the signed transaction:**
+    - Send the signed transaction to the desired blockchain network.
+    - Wait for the transaction to be mined and confirmed.
+
+4. **Verify the deployment:**
+    - Check the blockchain explorer to verify that the factory contract has been deployed to the expected address.
+    - Ensure the contract code matches the expected bytecode.
+
+By following these steps, you can deploy the factory contract to multiple blockchains, ensuring it has the same address on each one. This allows for consistent and deterministic deployment of other contracts using the factory contract.
+
+## **Popular Tools**
+
+1. [**Deterministic Deployment Proxy by Arachnid**](https://github.com/Arachnid/deterministic-deployment-proxy/tree/master)
+    - Signing the Proxy deployment tx is hidden, only recover the public key from signed tx.
+    - Somehow, the signing key and address is associated with the deployment tx. So if we change anything related to that deployment tx, the signing key and address will be different → proxy address is different.
+2. **Safe singleton Factory**
+    - Safe Multisig Wallet
+    - [Masa Finance](https://www.masa.ai/)
+    - Starlings lab forked and deployed it on many blockchains
+    - [align.network](http://align.network) - Long-term storage solution for Ethereum
+3. [**pcaversaccio/createx**](https://createx.rocks/)
+    
+    They offer two options for deploying [`CreateX`](https://github.com/pcaversaccio/createx/blob/main/src/CreateX.sol) to your desired chain:
+    
+    1. Deploy it yourself by using one of the pre-signed transactions. Details can be found in the subsequent paragraph.
+    2. Request a deployment by opening an [issue](https://github.com/pcaversaccio/createx/issues/new?assignees=pcaversaccio&labels=new+deployment+%E2%9E%95&projects=&template=deployment_request.yml&title=%5BNew-Deployment-Request%5D%3A+). You can significantly reduce the time to deployment by sending funds to cover the deployment cost (a reliable amount with a small tip 😏 would be ~0.3 ETH) to the deployer account: `0xeD456e05CaAb11d66C4c797dD6c1D6f9A7F352b5`.
+    3. See the [deployment result from Dune](https://dune.com/patronumlabs/createx)
+    
+4. [**pcaversaccio/xdeployer**](https://github.com/pcaversaccio/xdeployer) - hardhat
+    hardhat **plugin** to deploy your smart contracts across multiple EVM chains with the same deterministic address.
+    `npx hardhat xdeploy` 
+    
+    - It also deployed on Filecoin mainnet and testnet.
+    - contract creation transaction is the helper smart contract [`CreateX`](https://github.com/pcaversaccio/createx) with address `0xba5Ed099633D3B313e4D5F7bdc1305d3c28ba5Ed`
+    
+5. OpenZepplin Defender
+    - [How to tutorial](https://blog.openzeppelin.com/evm-deterministic-deployments-made-easy-with-openzeppelin-defender)
+
+Some of the most popular projects are using their own proxy for deterministic deployment across multiple chains. For example, 
```

Comment:
Is this something readers need to know for deploying factory contracts on Filecoin? If proxy is important, you might want to link an example to an external tutorial

Reply:
Deleted the 'some of the most popular...' portion.